pkg/iam/oidc/api.go

```diff
@@ -58,10 +59,17 @@ func NewOpenIDConnectManager(iamapi iamiface.IAMAPI, accountID, issuer string) (
 // CheckProviderExists will return true when the provider exists, it may return errors
 // if it was unable to call IAM API
 func (m *OpenIDConnectManager) CheckProviderExists() (bool, error) {
+
+	//check region partion
+	partion := "aws"
+	if strings.Contains(m.issuerURL.Host, "amazonaws.com.cn") {
+		partion = "aws-cn"
+	}
```

Comment:
The partition information is extracted in `ClusterProvider.NewOpenIDConnectManager` from the EKS cluster's ARN: https://github.com/weaveworks/eksctl/blob/de84cc0e11805eab79bba79208a41d23e8668af6/pkg/eks/eks.go#L233
Instead of extracting that information again from the issuer's host, WDYT about passing the partition to `iamoidc.NewOpenIDConnectManager`?

So this line https://github.com/weaveworks/eksctl/blob/de84cc0e11805eab79bba79208a41d23e8668af6/pkg/eks/eks.go#L243 becomes
```golang
return iamoidc.NewOpenIDConnectManager(c.Provider.IAM(), accountID, *c.Status.clusterInfo.cluster.Identity.Oidc.Issuer, parsedARN.Partition)
```